File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    a=[1,2,3,4,5]*2000
    b=[2,1,2,3,2,4,2,5]*1250
    c=[3,3,1,1,2,2,4,4,5,5]*1000
    cnt = [0,0,0]
    ans=[]
    idx=0
    for i in answers:
        cnt[0] = cnt[0]+1 if a[idx]==i else cnt[0]
        cnt[1] = cnt[1]+1 if b[idx]==i else cnt[1]
        cnt[2] = cnt[2]+1 if c[idx]==i else cnt[2]
        idx+=1
    
    for i in range(3):
        if cnt[i]==max(cnt): ans.append(i+1)
        
    return ans

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int n = answers.length;
        int[][] numbers = {
            {1,2,3,4,5},
            {2,1,2,3,2,4,2,5},
            {3,3,1,1,2,2,4,4,5,5}
        };

        int[] scores = {0, 0, 0};
        for(int i=0; i<n; i++){
            for (int j=0; j<3; j++){
                int k = i % numbers[j].length;
                if(answers[i] == numbers[j][k]){
                    scores[j] += 1;
                }
            }
        }
        
        int maxScore = Arrays.stream(scores).max().getAsInt();
        ArrayList<Integer> answer = new ArrayList<>();
        for (int i=0; i<3; i++){
            if(scores[i]==maxScore){
                answer.add(i+1);
            }
        }
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}